components/chef-run/lib/chef-run/startup.rb

```diff
@@ -0,0 +1,144 @@
+#
+# Copyright:: Copyright (c) 2018 Chef Software Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "chef-run/config"
+require "chef-run/text"
+require "chef-run/ui/terminal"
+require "chef-run/telemeter/sender"
+module ChefRun
+  class Startup
+    attr_reader :argv
+    T = ChefRun::Text.cli
+
+    def initialize(argv)
+      @term_init = false
+      @argv = argv.clone
+    end
+
+    def run
+      # Enable CLI output via Terminal. This comes first because we want to supply
+      # status output about reading and creating config files
+      init_terminal
+
+      # Some tasks we do only once in an installation:
+      first_run_tasks
+
+      # Call this every time, so that if we add or change ~/.chef-workstation
+      # directory structure, we can be sure that it exists. Even with a
+      # custom configuration, the .chef-workstation directory and subdirs
+      # are required.
+      setup_workstation_user_directories
+
+      # Startup tasks that may change behavior based on configuration value
+      # must be run after load_config
+      load_config
+
+      # Init logging using log level out of config
+      setup_logging
+
+      # Begin upload of previous session telemetry. (If telemetry is not enabled,
+      # in config the uploader will clean up previous session(s) without sending)
+      start_telemeter_upload
+
+      # Launch the actual chef-run behavior
+      start_chef_run
+    rescue ConfigPathInvalid => e
+      init_terminal
```

Comment:
Since we run `init_terminal` _first_ in the `run` method do we need to run it again here? Ditto to the other rescue below.

I realize this just ensures that our output call below will be successful, but it still feels redundant. Mostly it seems to protect us from a future situation where we are not calling `init_terminal` first in the `run` method